off-chain/src/HTTP/Types.purs

```diff
@@ -0,0 +1,89 @@
+module Seath.HTTP.Types where
+
+import Contract.Prelude
+import Undefined
+
+import Aeson
+  ( class DecodeAeson
+  , class EncodeAeson
+  , decodeJsonString
+  , encodeAeson
+  , fromString
+  , isString
+  , stringifyAeson
+  , toString
+  )
+import Ctl.Internal.Test.UtxoDistribution (encodeDistribution)
+import Data.Bifunctor (bimap)
+import Data.String
+  ( Pattern(Pattern)
+  , Replacement(Replacement)
+  , replace
+  , replaceAll
+  )
+import Data.UUID (UUID)
+import Payload.Client.DecodeResponse (class DecodeResponse)
+import Payload.Client.EncodeBody (class EncodeBody)
+import Payload.ContentType (class HasContentType, json)
+import Payload.ResponseTypes (Json(..), Response(..), ResponseBody(..))
+import Payload.Server.DecodeBody (class DecodeBody)
+import Payload.Server.Response (class EncodeResponse, encodeResponse, ok)
+import Seath.Core.Types (UserAction)
+
+-- Include action
+-- ! acrh: newtype wrapper made to avoid having DecodeBody/EncodeBody
+-- ! instances in `Network` layer, as it should be decoupled from HTTP layer
+newtype IncludeRequest a = IncludeRequest (UserAction a)
+
+derive instance Newtype (IncludeRequest a) _
+derive newtype instance Show a => Show (IncludeRequest a)
+
+instance decIncludeReq ::
+  ( DecodeAeson a
+  ) =>
+  DecodeBody (IncludeRequest a) where
+  decodeBody s = bimap show IncludeRequest $ decodeJsonString s
+
+instance encIncludeReq ::
+  ( EncodeAeson a
+  ) =>
+  EncodeBody (IncludeRequest a) where
+  encodeBody (IncludeRequest action) = show $ encodeAeson action
+
+instance includeContentType :: HasContentType (IncludeRequest a) where
+  getContentType _ = json
+
+newtype UID = UID UUID
+
+derive instance Newtype UID _
+
+instance showUID :: Show UID where
+  show (UID uuid) =
+    replace (Pattern "(UUID ") (Replacement "")
+      $ replace (Pattern ")") (Replacement "")
+      $ show uuid
+
+instance eaUID :: EncodeAeson UID where
+  encodeAeson = show >>> fromString
+
+-- Phantom types to make Spec and Handlers more readable
+-- Went with Record coz couldn't make `DecodeResponse` instance for custom `data` type
+type JSend :: forall err a. err -> a -> Type
+type JSend err a = { status :: String, data :: String }
+
+toJsend
```

Comment:
I vent with record and `toJsend` function because could not figure out if it possible to make required  instance for [DecodeResponse class](https://github.com/hoodunit/purescript-payload/blob/v0.4.0/src/Payload/Client/DecodeResponse.purs#L48-L79) for something like
```
data JSend
  = Success ...
   | Error ...
   | Failure ...
```

Reply:
The issue I couldn't figure out how to solve here is `DecodeResponse` instance that will be required for `payload`.
E.g.:
```newtype SomeReposne = SomeRsponse String

instance myTypeDecode :: DecodeResponse SomeReposne where
  decodeResponse _ = undefined
```
causes
```
[1/1 OverlappingInstances] src/HTTP/Types.purs:106:1

       v
  106  instance myTypeDecode :: DecodeResponse SomeReposne where
  107    decodeResponse _ = undefined
                                    ^
  
  Overlapping type class instances found for
  
    Payload.Client.DecodeResponse.DecodeResponse SomeReposne
  
  The following instances were found:
  
    Payload.Client.DecodeResponse.decodeResponseDefault
    Seath.HTTP.Types.myTypeDecode
  
  in type class instance
  
    Payload.Client.DecodeResponse.DecodeResponse SomeReposne
    
```